project/versioning.scala

```diff
@@ -0,0 +1,37 @@
+object Versioning {
+
+  import java.time._
+  import sbt._
+  import sbt.Keys._
+
+  def zeroPadded(upto: Int) =
+    (s: String) => (("0" * upto) ++ s.take(upto)) takeRight upto
+
+  //currently allows "-SNAPSHOT" as a valid suffix for a release
+  val releasePattern = raw"^ci-release-(\d+)(-.*)?".r
+
+  /* implement the logic for versioning explained in the build file
+   * the tag parameter should have a format like:
+   *   ci-release-1-3-d3a9863 <-- this latter part is the result of `git describe`
+   *              ^
+   *              |- this is the version we care about
+   */
+  def generate(major: Int, date: LocalDate, tag: String): String = {
```

Comment:
Can we covert the comments to ScalaDoc format?

Reply:
done